tests/apidoc_login_test.go

```diff
@@ -81,6 +82,70 @@ func TestLogin(t *testing.T) {
 				},
 			})
 		})
+		// sytest: POST /login can log in as a user with just the local part of the id
+		t.Run("POST /login can log in as a user with just the local part of the id", func(t *testing.T) {
+			t.Parallel()
+			createDummyUser(t, unauthedClient, "@test_user:matrix")
```

Comment:
`createDummyUser` will pass whatever you give it as a `userID` into the request as `username`, which is treated as the user localpart. This means we don't need `@` or `:matrix` here.

Reply:
Also, `:matrix` wouldn't be valid. It has to be the domain of the server, which for these tests is `:hs1`.

Reply:
Hi @Kegsay I just modified the following test to not use createDummyUser and do the stuff we have been doing in the perl test
```go
			res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "register"}, json.RawMessage(`{
				"auth": {
					"type": "m.login.dummy"
				},
				"username": "local-login-user",
				"password": "superuser"
			}`))
			body, _ := ioutil.ReadAll(res.Body)
			userID := gjson.Get(string(body), "user_id")

			reg := regexp.MustCompile("@([^:]*):")
			localPart := reg.FindStringSubmatch(userID.Str)
			res = unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
				"type": "m.login.password",
				"identifier": {
					"type": "m.id.user",
					"user": "`+localPart[1]+`"
				},
				"password": "superuser"
			}`))
```

I have got the userID from the response of the register request and then trimmed it as was done in sytest, and tests are passing now too. Does this looks fine 🤔 ?